include/merlin/memory_pool.cuh

```diff
@@ -0,0 +1,334 @@
+/*
```

Comment:
Could you add some required UT cases for this file?

Reply:
Done!